### PR TITLE
DownloadUrl at the root of a bucket throws proper error message

### PR DESCRIPTION
**Pull request by fredzqm**
#429

StorageReference.getDownloadUrl will throw if called on a bucket root reference.

### CODE REVIEW

**Comment by fredzqm** [comment]
@schmidt-sebastian

I got this hacked together without the dev setup.
Please take a look, it should be simple, but addresses many people's confusion over the error message.

See #429

File: firebase-storage/src/main/java/com/google/firebase/storage/GetDownloadUrlTask.java [hunk 1]
```diff
@@ -43,7 +43,9 @@
 
     this.storageRef = storageRef;
     this.pendingResult = pendingResult;
-
+    if (storageRef.getRoot().getName().equals(storageRef.getName())) {
+      throw new IllegalArgumentException("Cannot get download url at the root of a bucket.");
```
Comment (schmidt-sebastian) [inline firebase-storage/src/main/java/com/google/firebase/storage/GetDownloadUrlTask.java]: Can you change this to use the method name (getDownloadUrl())? Furthermore, does the same problem exists if a user calls getMetadata()?

Comment (schmidt-sebastian) [reply firebase-storage/src/main/java/com/google/firebase/storage/GetDownloadUrlTask.java]: Can you make the error message something like: "getDownloadUrl() is not supported at the root of the Storage bucket."

File: firebase-storage/src/main/java/com/google/firebase/storage/GetMetadataTask.java [hunk 2]
```diff
@@ -41,6 +41,9 @@
 
     this.mStorageRef = storageRef;
     this.mPendingResult = pendingResult;
+    if (storageRef.getRoot().getName().equals(storageRef.getName())) {
+      throw new IllegalArgumentException("Cannot get object metedata at the root of a bucket.");
+    }
```
Comment (schmidt-sebastian) [inline firebase-storage/src/main/java/com/google/firebase/storage/GetMetadataTask.java]: Can you make the error message something like: "getMetadata() is not supported at the root of the Storage bucket."

Comment (fredzqm) [reply firebase-storage/src/main/java/com/google/firebase/storage/GetMetadataTask.java]: Done

File: firebase-storage/src/main/java/com/google/firebase/storage/GetDownloadUrlTask.java (hunk 1 above)
Comment (fredzqm) [reply firebase-storage/src/main/java/com/google/firebase/storage/GetDownloadUrlTask.java]: Done

File: firebase-storage/CHANGELOG.md [hunk 3]
```diff
@@ -4,3 +4,6 @@
   and are not mentioned in our documentation.
 - [internal] Updated the SDK initialization process and removed usages of
   deprecated methods.
+- [changed] Added validation to `StorageReference.getDownloadUrl` and 
+  `StorageReference.getMetadata` to return error if the reference is the root
```
Comment (schmidt-sebastian) [inline firebase-storage/CHANGELOG.md]: s/to return error/to return an error/ :)

**Pull request merged**